### PR TITLE
[SQL] Bug fix for UMI

### PR DESCRIPTION
**Pull request by viparek**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Sql/Sql/Common/ResourceIdentityHelper.cs [hunk 1]
```diff
@@ -142,5 +151,28 @@ public static Management.Sql.Models.ResourceIdentity GetIdentityObjectFromType(b
             return identityResult;
 
         }
+
+        public static string GetEnumValue(ResourceIdentityType resourceIdentityType)
+        {
```
Comment (BethanyZhou) [inline src/Sql/Sql/Common/ResourceIdentityHelper.cs:156]: Looks like this method is never used, please have a look

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 2]
```diff
@@ -348,9 +348,9 @@ public class NewAzureSqlManagedInstance : ManagedInstanceCmdletBase
         /// Type of identity to be assigned to the server..
         /// </summary>
         [Parameter(Mandatory = false,
-            HelpMessage = "Type of Identity to be used. Possible values are SystemAsssigned, UserAssigned, SystemAssignedUserAssigned and None.")]
-        [PSArgumentCompleter("SystemAssigned", "UserAssigned", "SystemAssignedUserAssigned", "None")]
-        public string IdentityType { get; set; }
+            HelpMessage = "Type of Identity to be used. Possible values are SystemAssigned, UserAssigned, 'SystemAssigned,UserAssigned' and None.")]
+        [PSArgumentCompleter("SystemAssigned", "UserAssigned", "SystemAssigned,UserAssigned", "None")]
+        public Object[] IdentityType { get; set; }
```
Comment (BethanyZhou) [inline src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs:353]: Object[] is not allowed in IdentityType either. Delete SystemAssignedUserAssigned will cause breaking change

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/Common/ResourceIdentityHelper.cs [hunk 3]
```diff
@@ -30,10 +30,11 @@ public enum ResourceIdentityType
 
     public class ResourceIdentityHelper
     {
-        public static Management.Sql.Models.ResourceIdentity GetIdentityObjectFromType(bool assignIdentityIsPresent, System.Object[] resourceIdentityType, List<string> userAssignedIdentities, Management.Sql.Models.ResourceIdentity existingResourceIdentity)
+        public static Management.Sql.Models.ResourceIdentity GetIdentityObjectFromType(bool assignIdentityIsPresent, string resourceIdentityType, List<string> userAssignedIdentities, Management.Sql.Models.ResourceIdentity existingResourceIdentity)
         {
             Management.Sql.Models.ResourceIdentity identityResult = null;
-            string resourceIdType = null;
+            string resourceIdType = resourceIdentityType;
+            /*
```
Comment (BethanyZhou) [inline src/Sql/Sql/Common/ResourceIdentityHelper.cs:37]: Please remove commented lines

**Review by BethanyZhou — COMMENTED** [review]
Hi @viparek , Can `New-AzSqlInstance -IdentityType` handle `SystemAssignedUserAssigned` now?  I mean even we don't provide `SystemAssignedUserAssigned` in `ArgumentCompleter` anymore, we should be able to handle it in our logic to avoid breaking change. e.g. previous PS script can work as before.

**Comment by viparek** [comment]
> Hi @viparek , Can `New-AzSqlInstance -IdentityType` handle `SystemAssignedUserAssigned` now? I mean even we don't provide `SystemAssignedUserAssigned` in `ArgumentCompleter` anymore, we should be able to handle it in our logic to avoid breaking change. e.g. previous PS script can work as before.

It is by design that we are not supporting 'SystemAssignedUserAssigned`. We do not want to create confusion with the customers by adding both. Also like we discussed earlier, IdentityType is a brand new param added in the last PS release, and the feature it supports is also not released yet. So this change will not have any impact.

**Comment by BethanyZhou** [comment]
> It is by design that we are not supporting 'SystemAssignedUserAssigned`. We do not want to create confusion with the customers by adding both. Also like we discussed earlier, IdentityType is a brand new param added in the last PS release, and the feature it supports is also not released yet. So this change will not have any impact.

Yes, I know we should respect the design in swagger, handle 'SystemAssignedUserAssigned' in PS logic is a workaround to avoid breaking change. Please reply email to track this discussion about the workaround, thanks.

One question, for "SystemAssigned,UserAssigned", is it should be "SystemAssigned, UserAssigned"? Is there any blank between ""SystemAssigned," and "UserAssigned" in swagger? I saw "SystemAssigned, UserAssigned" in other swaggers.

**Review by BethanyZhou — APPROVED** [review]

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Please update the description in [New-AzSqlServer.md](https://github.com/Azure/azure-powershell/blob/ab497db062ef15f3b8b07f0aa02055d780aea1a3/src/Sql/Sql/help/New-AzSqlServer.md#-identitytype) and [New-AzSqlInstance.md](https://github.com/Azure/azure-powershell/blob/ab497db062ef15f3b8b07f0aa02055d780aea1a3/src/Sql/Sql/help/New-AzSqlInstance.md#-identitytype) as well

**Review by BethanyZhou — DISMISSED** [review]

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**